drivers/adc/ad7124/ad7124.h

```diff
@@ -247,6 +249,9 @@
 #define AD7124_CFG_REG_REF_SEL(x)  ((x) & 0x3) << 3
 #define AD7124_CFG_REG_PGA(x)      (((x) & 0x7) << 0)
 
+/* PGA Gain Value */
+#define AD7124_PGA_GAIN(x)     (1 << (x))
```

Comment:
why not just use AD7124_CFG_REG_PGA for PGA configuration. 

This definition will only allow gain: 1, 2, 4, 16.
Check page 90 of https://www.analog.com/media/en/technical-documentation/data-sheets/AD7124-8.pdf

Reply:
The intent of AD7124_PGA_GAIN is to provide a method to calculate the actual gain values (1, 2, 4, 8, 16, 32, 64, 128) that correspond to the 3-bit config values stored in the PGA bit field of the CFG_REG. The AD7124_PGA_GAIN 'function' is useful when converting raw DATA register values to a voltage.